--- a/src/meson.build
+++ b/src/meson.build
@@ -2,9 +2,9 @@
 subdir('spirv')
 subdir('vulkan')
 subdir('dxvk')
-subdir('dxgi')
-subdir('dxbc')
-subdir('d3d11')
-subdir('d3d10')
+subdir('dxgi')
+#subdir('dxbc')
+#subdir('d3d11')
+#subdir('d3d10')
 subdir('dxso')
 subdir('d3d9')
--- a/src/dxgi/meson.build
+++ b/src/dxgi/meson.build
@@ -19,7 +19,7 @@
   name_prefix         : '',
   dependencies        : [ dxvk_dep ],
   include_directories : dxvk_include_path,
-  install             : true,
+  build_by_default    : false,
   vs_module_defs      : 'dxgi'+def_spec_ext,
   objects             : not dxvk_msvc ? 'dxgi'+def_spec_ext : [],
   override_options    : ['cpp_std='+dxvk_cpp_std])
