--- a/dev/null
+++ b/meson.build
@@ -0,0 +1,9 @@
+project('libstrangle', ['c'], default_options : [ 'c_std=c99', 'c_link_args=["-Wl,-z,relro,-z,now"]' ])
+
+add_project_arguments('-DHOOK_DLSYM', language : 'c')
+add_project_arguments('-rdynamic', '-fvisibility=hidden', language : 'c')
+
+lib_dl   = meson.get_compiler('c').find_library('dl')
+lib_rt   = meson.get_compiler('c').find_library('rt')
+
+subdir('src')
--- a/dev/null
+++ b/src/meson.build
@@ -0,0 +1,12 @@
+libstrangle_src = [
+  'egl.c',
+  'gl.c',
+  'glx.c',
+  'libstrangle.c',
+  'real_dlsym.c',
+  'vulkan.c',
+]
+
+libstrangle_deps = [ lib_dl, lib_rt ]
+
+shared_library('strangle', libstrangle_src, dependencies: libstrangle_deps, install: true)
