diff --git a/meson.build b/meson.build
index c3cc5f5..2cf1c7c 100644
--- a/meson.build
+++ b/meson.build
@@ -65,5 +65,5 @@ dxup_version = vcs_tag(
 subdir('src')
 
 if dxup_compiler.get_id() != 'msvc'
-  subdir('utils')
+#  subdir('utils')
 endif
diff --git a/src/meson.build b/src/meson.build
index fa859d2..a281fa7 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -2,5 +2,5 @@ subdir('util')
 subdir('dx9asm')
 subdir('d3d9')
 
-subdir('d3d10_1')
-subdir('dxgi')
\ No newline at end of file
+#subdir('d3d10_1')
+#subdir('dxgi')
\ No newline at end of file
