diff --git a/meson.build b/meson.build
index 23e848e..ada06d1 100644
--- a/meson.build
+++ b/meson.build
@@ -62,12 +62,10 @@ dxvk_version = vcs_tag(
 
 subdir('src')
 
-enable_tests = get_option('enable_tests')
-
-if enable_tests
+if get_option('enable_tests')
   subdir('tests')
 endif
 
-if dxvk_compiler.get_id() != 'msvc'
+if get_option('enable_utils')
   subdir('utils')
 endif
diff --git a/meson_options.txt b/meson_options.txt
index 6e0f05b..17d06be 100644
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -1 +1,2 @@
 option('enable_tests', type : 'boolean', value : false)
+option('enable_utils', type : 'boolean', value : false)
